pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -101,36 +99,17 @@ func getProjectAndZone() (string, string, error) {
 	return projectID, zone, nil
 }
 
-func getInstanceID() (string, error) {
-	result, err := metadata.Get("instance/hostname")
+func getNetworkName(svc *compute.Service, projectID string) (string, error) {
+	networkList, err := svc.Networks.List(projectID).Do()
```

Comment:
It is going to be less reliable for machines running in GCE to hit the GCE API than the local metadata server. In the past we've seen it do things like exhaust the quota limit for the user making it impossible for them to use their project for a day.